src/test/unit/VoteWeigherBaseUnit.t.sol

```diff
@@ -0,0 +1,664 @@
+        // make sure the quorum strategies and weights haven't changed
+        for (uint i = 0; i < strategiesAndWeightingMultipliers.length; i++) {
+            IVoteWeigher.StrategyAndWeightingMultiplier memory strategyAndWeightingMultiplier = voteWeigher.strategyAndWeightingMultiplierForQuorumByIndex(quorumNumber, i);
+            assertEq(address(strategyAndWeightingMultiplier.strategy), address(strategiesAndWeightingMultipliers[i].strategy));
+            assertEq(strategyAndWeightingMultiplier.multiplier, strategiesAndWeightingMultipliers[i].multiplier);
+        }
+    }
+
+    function testModifyStrategyWeights_NotFromServiceManagerOwner_Reverts(
+        address notServiceManagerOwner,
+        IVoteWeigher.StrategyAndWeightingMultiplier[] memory strategiesAndWeightingMultipliers,
+        uint256[] memory strategyIndices,
+        uint96[] memory newWeights
+    ) public {
+        cheats.assume(notServiceManagerOwner != serviceManagerOwner);
+        strategiesAndWeightingMultipliers = _convertToValidStrategiesAndWeightingMultipliers(strategiesAndWeightingMultipliers);
+
+        // create a valid quorum
+        uint8 quorumNumber = voteWeigher.quorumCount();
+        cheats.prank(serviceManagerOwner);
+        voteWeigher.createQuorum(strategiesAndWeightingMultipliers);
+
+        // modify certain strategies
+        cheats.prank(notServiceManagerOwner);
+        cheats.expectRevert("VoteWeigherBase.onlyServiceManagerOwner: caller is not the owner of the serviceManager");
+        voteWeigher.modifyStrategyWeights(quorumNumber, strategyIndices, newWeights);
+    }
+
+    function testModifyStrategyWeights_Valid(
+        IVoteWeigher.StrategyAndWeightingMultiplier[] memory strategiesAndWeightingMultipliers,
+        uint256[] memory strategyIndices,
+        uint96[] memory newWeights
+    ) public {
+        strategiesAndWeightingMultipliers = _convertToValidStrategiesAndWeightingMultipliers(strategiesAndWeightingMultipliers);
+
+        // take indices modulo length
+        for (uint256 i = 0; i < strategyIndices.length; i++) {
+            strategyIndices[i] = strategyIndices[i] % strategiesAndWeightingMultipliers.length;
+        }
+        strategyIndices = _removeDuplicatesUint256(strategyIndices);
+        cheats.assume(strategyIndices.length > 0);
+        cheats.assume(strategyIndices.length < strategiesAndWeightingMultipliers.length);
+
+        // trim the provided weights to the length of the strategyIndices
+        uint96[] memory newWeightsTrim = new uint96[](strategyIndices.length);
+        for (uint256 i = 0; i < strategyIndices.length; i++) {
+            if(i < newWeights.length) {
+                newWeightsTrim[i] = newWeights[i];
+            } else {
+                newWeightsTrim[i] = strategiesAndWeightingMultipliers[strategyIndices[i]].multiplier - 1;
+            }
+        }
+        newWeights = newWeightsTrim;
```

Comment:
so this is just making the lengths match. seems OK, but I might describe it more accurately (it's trimming in some cases and extending in others)

Reply:
fixed